src/transformers/integrations.py

```diff
@@ -337,6 +341,74 @@ def run_hp_search_sigopt(trainer, n_trials: int, direction: str, **kwargs) -> Be
     return best_run
 
 
+def run_hp_search_wandb(trainer, n_trials: int, direction: str, **kwargs) -> BestRun:
+    from .integrations import is_wandb_available
+
+    assert is_wandb_available(), "This function needs wandb installed: `pip " "install wandb`"
+    import wandb
+
+    # add WandbCallback if not already added in trainer callbacks
+    reporting_to_wandb = False
+    for callback in trainer.callback_handler.callbacks:
+        if isinstance(callback, WandbCallback):
+            reporting_to_wandb = True
+            break
+    if not reporting_to_wandb:
+        trainer.add_callback(WandbCallback())
+    trainer.args.report_to = "wandb"
+    best_trial = {"run_id": None, "objective": None, "hyperparameters": None}
+    sweep_id = kwargs.pop("sweep_id", None)
+    project = kwargs.pop("project", None)
+    name = kwargs.pop("name", None)
+    entity = kwargs.pop("entity", None)
+    metric = kwargs.pop("metric", None) or "eval/loss"
```

Comment:
Why not put it directly as a default value?
```suggestion
    metric = kwargs.pop("metric", "eval/loss")
```

Reply:
lol yes! will do